generator/sbpg/targets/resources/sbp_messages_mod.rs

```diff
@@ -22,26 +23,49 @@ use self::(((p.identifier|mod_name)))::(((m.identifier|camel_case)));
 ((*- endfor *))
 use self::unknown::Unknown;
 
-use crate::serialize::SbpSerialize;
+mod lib {
+    //! Common imports so we can just `use super::lib::*` in all the message files
 
-pub trait SbpMessage: SbpSerialize {
+    pub use std::convert::{TryFrom, TryInto};
+
+    pub use crate::wire_format::{WireFormat, PayloadParseError};
+    pub use crate::sbp_string::{SbpString, Unterminated, NullTerminated, Multipart, DoubleNullTerminated};
+
+    #[cfg(feature = "swiftnav-rs")]
+    pub use crate::time;
+
+    pub use super::{ConcreteMessage, Sbp, SbpMessage, TryFromSbpError};
+}
+
+use lib::*;
+
+/// Common functionality available to all SBP messages.
+pub trait SbpMessage: WireFormat + Clone + Sized {
+    /// Get the message name.
     fn message_name(&self) -> &'static str;
+    /// Get the message type.
     fn message_type(&self) -> u16;
+    /// Get the sender_id if it is set.
     fn sender_id(&self) -> Option<u16>;
+    /// Set the sender id.
     fn set_sender_id(&mut self, new_id: u16);
-    fn to_frame(&self) -> std::result::Result<Vec<u8>, crate::FramerError>;
-    fn write_frame(&self, buf: &mut Vec<u8>) -> std::result::Result<(), crate::FramerError>;
+    /// Get the GPS time associated with the message.
     #[cfg(feature = "swiftnav-rs")]
-    fn gps_time(&self) -> Option<std::result::Result<crate::time::MessageTime, crate::time::GpsTimeError>> {
+    fn gps_time(&self) -> Option<Result<crate::time::MessageTime, crate::time::GpsTimeError>> {
         None
     }
 }
 
-pub trait ConcreteMessage: SbpMessage + std::convert::TryFrom<Sbp, Error = TryFromSbpError> + Clone + Sized {
+/// Implemented by messages who's message name and type are known at compile time.
+/// This is everything that implements [SbpMessage] except for [Sbp].
+pub trait ConcreteMessage: SbpMessage + TryFrom<Sbp, Error = TryFromSbpError> {
```

Comment:
If `ConcreteMessage` is only for use in the sbp crate, should this be marked as `pub(crate)`?

Reply:
Hmm it can't be marked private because it a appears in `Event` which needs to be public. Could maybe add `#[doc(hidden)]`. Depends on if we think having access to message names/ids at compile time is useful. Right now this is the only way to get that